site/src/components/WorkspaceStatusBadge/WorkspaceStatusBadge.tsx

```diff
@@ -87,22 +89,49 @@ export const getStatus = (buildStatus: WorkspaceBuild["status"]) => {
 }
 
 export type WorkspaceStatusBadgeProps = {
-  build: WorkspaceBuild
+  workspace: Workspace
   className?: string
 }
 
+const ImpendingDeletionBadge: FC<
+  PropsWithChildren<Partial<WorkspaceStatusBadgeProps>>
+> = ({ className }) => {
+  const { entitlements, experiments } = useDashboard()
+  const allowAdvancedScheduling =
+    entitlements.features["advanced_template_scheduling"].enabled
+  // This check can be removed when https://github.com/coder/coder/milestone/19
+  // is merged up
+  const allowWorkspaceActions = experiments.includes("workspace_actions")
+
+  if (!allowAdvancedScheduling || !allowWorkspaceActions) {
+    return null
+  }
```

Comment:
Since we are already checking if we should or not display the badge on `displayImpendingDeletion` I think we should move those checks to be there since they look doing the same thing.

Reply:
@BrunoQuaresma those checks rely on a hook, which has to be called in a component. Would you rather I call the hook in the `WorkspaceStatusBadge` component? I definitely could; I just didn't want to screw up all the storybook stories.

Alternatively, I thought about calling this hook in `WorkspacesPage` and then passing down props from there - but there are like 7 layers of prop passing and it just seemed like a lot to me.

Reply:
The thing for me is to have the "should display or not" check in one single place. Splitting this check into different places looks confusing to me. But if it is hard, I would add a comment so that in the future, if we want to change the display behavior, we know there are two places to look for.

Reply:
Agreed - as the ideal behavior is to have the check in one place, I think the best path forward may be to call the hook in the badge component and then ignore Storybook's complaints, rather than let our testing tool dictate how we code.
I'll adjust.